src/constants.ts

```diff
@@ -0,0 +1 @@
+export const STALE_TIMEOUT = 10000;
```

Comment:
10 seconds might be on the short side, as I could see an extreme request to the text editor actually taking that long, maybe 15, 20, or 30 would be a bit nicer.
Probably also suffix the name with _MS so the units are obvious here.